File: paper_artifacts/experiments/llr8/artifacts/sources/oss120b-fortran-skills/tsvc_2_s232__submitted.f90
subroutine tsvc_2_s232_fp64(aa, bb, len_2d) bind(C, name="tsvc_2_s232_fp64")
  use iso_c_binding
  integer(c_int64_t), value, intent(in) :: len_2d
  real(c_double), intent(inout) :: aa(len_2d, len_2d)
  real(c_double), intent(in) :: bb(len_2d, len_2d)
  integer(c_int64_t) :: i, j

  !$omp parallel do private(i) schedule(static)
  do j = 2, len_2d
    do i = 2, j
      aa(i, j) = aa(i-1, j) * aa(i-1, j) + bb(i, j)
    end do
  end do
  !$omp end parallel do

end subroutine tsvc_2_s232_fp64

File: paper_artifacts/experiments/llr9/artifacts/sources/oss120b-c-skills/tsvc_2_s232__submitted.c
#include <stdint.h>
#include <omp.h>

void tsvc_2_s232_fp64(double *restrict aa, const double *restrict bb, const int64_t LEN_2D) {
    #pragma omp parallel for schedule(static)
    for (int64_t j = 1; j < LEN_2D; ++j) {
        for (int64_t i = 1; i <= j; ++i) {
            double prev = aa[j * LEN_2D + (i - 1)];
            aa[j * LEN_2D + i] = prev * prev + bb[j * LEN_2D + i];
        }
    }
}
